src/pygrep_hooks/plugins.rs

```diff
@@ -0,0 +1,54 @@
+use rustpython_ast::{Expr, ExprKind};
+
+use crate::ast::types::{BindingKind, Range};
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind};
+
+/// PGH002 - deprecated use of logging.warn
+pub fn deprecated_log_warn(checker: &mut Checker, func: &Expr) {
+    match &func.node {
+        ExprKind::Attribute { attr, value, .. } => {
```

Comment:
Is it possible to write these instead using the `call_path` utilities? Like `let call_path = dealias_call_path(collect_call_paths(expr), import_aliases);` and such that we have elsewhere? These tend to do a better job of tracking aliased imports. (The downside is they might raise false-positives when you redefine `warn`, but that feels less important to me...)

Reply:
I'm having problems understanding this one or rather, seeing how this helps readability.

I would still have to handle at least two cases to lookup how something was imported.

```rust
let call_path = dealias_call_path(collect_call_paths(func), &checker.import_aliases);
    match &call_path[..] {
        [name] => ..,
        [module, name] => ..,
        _ => (),
    }
```

Otherwise something like `from warnings import warn; warn("hello")` would trigger the check.

Reply:
Yeah, you _will_ need a dedicated check for `log.warn` since that's not based on imports at all.

But I think the check can be this? Unless I'm misunderstanding some of the intent:

```rs
let call_path = dealias_call_path(collect_call_paths(expr), import_aliases);

if match_call_path(&call_path, "logging", "warn") {
  // Not ok
} else if call_path == &["log", "warn"] {
  // Not ok
}
```

This would avoid ever flagging on any form of `warnings.warn`.

Reply:
What about `from warnings import warn; warn("hello");`?  In my tests, `dealias_call_path` would just give me back `["warn"]`, which doesn't tell me anything about where it comes from. It's one of the testcases of pygrep-hooks.  I do agree with your solution if we just look at `logging.warn`& `log.warn`.

Reply:
Thank you, I didn't know about `match_call_path`. My confusion stem from the fact that just `dealias_call_path` would give me `["warn"]`in some cases, which is not enough information.